x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_logic.ts

```diff
@@ -87,6 +90,14 @@ export const EngineLogic = kea<MakeLogicType<EngineValues, EngineActions>>({
       () => [selectors.engine],
       (engine) => engine?.unconfirmedFields?.length > 0,
     ],
+    searchKey: [
```

Comment:
We need to POST the search key to the new search experience URL from the Search UI form. Since all keys that are valid for a particular engine are already available on `engine.apiTokens` from state, I added a selector to easily access the first available search token.

Since tokens that do not have access to the current engine are not returned in this object, I do not have to check their access level, as we already know it has access to the current engine.

Reply:
I really like this added logic/selector. Feels elegant and makes sense.